### PR TITLE
:seedling: Run E2E PAT test for push to main

### PR DESCRIPTION
**Pull request by naveensrinivasan**
- Add E2E PAT tests for push to main.

#### What kind of change does this PR introduce?

(Is it a bug fix, feature, docs update, something else?)

- [ ] PR title follows the guidelines defined in our [pull request documentation](https://github.com/ossf/scorecard/blob/main/CONTRIBUTING.md#pr-process)

#### What is the new behavior (if this is a feature change)?**

- [ ] Tests for the changes have been added (for bug fixes/features)

#### Does this PR introduce a user-facing change?

For user-facing changes, please add a concise, human-readable release note to
the `release-note`

(In particular, describe what changes users might need to make in their
application as a result of this pull request.)

```release-note

```

### CODE REVIEW

File: .github/workflows/integration-main.yml [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    steps:
+      - name: Harden Runner
+        uses: step-security/harden-runner@128a63446a954579617e875aaab7d2978154e969 # v1
+        with:
+          egress-policy: audit # TODO: change to 'egress-policy: block' after couple of runs
+
+      - name: pull_request actions/checkout
+        uses: actions/checkout@8e5e7e5ab8b370d6c329ec480221332ada57f0ab # v2.3.4
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
```
Comment (spencerschrock) [inline .github/workflows/integration-main.yml:51]: I don't think this would be defined?

File: .github/workflows/integration-main.yml [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+      - name: setup-go
+        uses: actions/setup-go@fac708d6674e30b6ba41289acaab6d4b75aa0753 # v2.2.0
+        with:
+          go-version: '1.19'
```
Comment (spencerschrock) [inline .github/workflows/integration-main.yml:56]: Let's start standardizing on `go-version-file` like done [here](https://github.com/ossf/scorecard/blob/f8d33f8b3ef2ef6502330ac1d4bf51f592fd6c95/.github/workflows/publishimage.yml#L43-L47).

File: .github/workflows/integration-main.yml [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+        env:
+          GITHUB_AUTH_TOKEN: ${{ secrets.GH_AUTH_TOKEN }}
+        with:
+          max_attempts: 3
+          retry_on: error
+          timeout_minutes: 30
+          command: make e2e-pat
+
+      - name: codecov
+        uses: codecov/codecov-action@eaaf4bedf32dbdc6b720b63067d99c4d77d6047d # 2.1.0
```
Comment (spencerschrock) [inline .github/workflows/integration-main.yml:75]: would this upload interfere with the one from `main.yml`?

**Review by spencerschrock — COMMENTED** [review]
I support the idea, but is there any reason not to just add this here?
https://github.com/ossf/scorecard/blob/f8d33f8b3ef2ef6502330ac1d4bf51f592fd6c95/.github/workflows/main.yml#L66-L72

This would let us avoid the extra setup steps/`go mod` downloads

**Comment by naveensrinivasan** [comment]
> I support the idea, but is there any reason not to just add this here?
> 
> https://github.com/ossf/scorecard/blob/f8d33f8b3ef2ef6502330ac1d4bf51f592fd6c95/.github/workflows/main.yml#L66-L72
> 
> This would let us avoid the extra setup steps/`go mod` downloads

I agree. Fixed it. PTAL

**Review by spencerschrock — APPROVED** [review]

**Pull request merged**